src/prover_disk.hpp

```diff
@@ -46,11 +46,11 @@ struct plot_header {
 // of space, for a given challenge.
 class DiskProver {
 public:
-    // The costructor opens the file, and reads the contents of the file header. The table pointers
+    // The constructor opens the file, and reads the contents of the file header. The table pointers
     // will be used to find and seek to all seven tables, at the time of proving.
-    explicit DiskProver(std::string filename)
+    explicit DiskProver(const std::string& filename)
     {
-        struct plot_header header;
+        struct plot_header header{};
         this->filename = filename;
```

Comment:
if you instead would move the filename into the member variable here, callers of this function would be able to hand over ownership of the string, resulting in not copying the string at all. I think doing that is more ideomatic C++ these days, but it's not like it makes a difference here anyway.